internal/oci/runtime_oci.go

```diff
@@ -232,6 +232,14 @@ func (r *runtimeOCI) StartContainer(c *Container) error {
 	c.opLock.Lock()
 	defer c.opLock.Unlock()
 
+	// if the annotation cpu-load-balancing.kubernetes.io=true present under the container
```

Comment:
I wonder if cri-o should be doing some sanity checks too. As this is written if will allow any Pod to disable cpu balancing even if it is not the sole inhabitant of the cpu. That might disrupt other unrelated workloads.

In k8s world this should only be allowed when the Pod belongs to the guaranteed resource class. Meaning it is the only Pod using the requested cpus.

Reply:
CRI-O is generally pretty dumb, and listens to whatever the kubelet tells it to do. We'd need some admission controller to protect against this

Reply:
I'm just saying this is unsafe by default as it is based on an annotation that anyone can set.